packages/flutter/lib/src/foundation/persistent_hash_map.dart

```diff
@@ -0,0 +1,392 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/// A collection of key/value pairs, from which you can retrieve a value
+/// using its associated key.
+///
+/// This class implements a persistent map: extending this map with a new
```

Comment:
'persistent' makes me think that the map survives the program restarting.
The map is 'functional' in the sense of 'functional programming' - the map is an unchanging value rather than a mutatable entity. Any statefulness happens in the client of the library.

Reply:
I think _persistent data structure_ is a [fairly standard terminology](https://en.wikipedia.org/wiki/Persistent_data_structure). It means that the previous version of the data-structure persists after "modification".